lib/Transforms/IPO/StructFieldCacheAnalysis.cpp

```diff
@@ -385,18 +472,30 @@ static void performIRAnalysis(Module &M,
   DEBUG_WITH_TYPE(DEBUG_TYPE_IR, StructManager->debugPrintAnnotatedModule());
 }
 
+static void applyFilters(StructFieldAccessManager *StructManager) {
+  StructManager->summarizeFunctionCalls();
+  StructManager->applyFiltersToStructs();
+  StructManager->printStats();
+}
+
 static bool performStructFieldCacheAnalysis(
-    Module &M, function_ref<BlockFrequencyInfo *(Function &)> LookupBFI) {
+    Module &M, function_ref<BlockFrequencyInfo *(Function &)> LookupBFI,
+    function_ref<BranchProbabilityInfo *(Function &)> LookupBPI) {
   DEBUG(dbgs() << "Start of struct field cache analysis\n");
-  StructFieldAccessManager StructManager(M, LookupBFI);
+  StructFieldAccessManager StructManager(M, LookupBFI, LookupBPI);
   // Step 0 - retrieve debug info for all struct TODO: disable for now because
   // it's not supporting annonymous structs
-  // StructManager.retrieveDebugInfoForAllStructs();
   // Step 1 - perform IR analysis to collect info of all structs
   performIRAnalysis(M, &StructManager);
-  StructManager.summarizeFunctionCalls();
-  StructManager.applyFiltersToStructs();
-  StructManager.printStats();
+  // Step 2 - summarize function calls and apply filters
+  applyFilters(&StructManager);
```

Comment:
I intended to also apply the filter when only perform IR because I print stats after applying filters. I'll change the flag name to PerformCodeAnalysisOnly to avoid the confusion...